docs/git/branches.md

```diff
@@ -0,0 +1,169 @@
+
+Running this command doesn't change any of the files in our repository - they stay the same, but if we commit changes, they will only get committed on this branch. If we switched back to the `main` branch, the changes would be un-done. But if we switch back to `story`, they'll come back!
+
+### How do I know what branch I'm on?
+
+You've already run one command which tells you this, you just may not have noticed!
+
+```console
+$ git status
+On branch story
+nothing to commit, working tree clean
+```
+
+That's right, `git status` tells you what branch you're on (and about any changes you haven't committed).
+
+You can also find out more about branches by running:
+
+```console
+$ git branch
+  main
+* story
+```
+
+`git branch` tells you what branches exist in your repository, and puts a star next to the one you're currently switched to.
+
+### Exercise 1 (15 minutes)
+
+Let's make a new branch, and make a new commit on it.
+
+Create a branch named story, and edit `mug-brownie.md`.
+
+Before the "What you need" heading, make up a story about how the author discovered the recipe, or a time they enjoyed it.
+
+When you've written a bit, make a commit (make sure to give it a good commit message!)
+
+Explore the history with `git log` - make sure you understand the commits in your history and why they're there.
+
+### Switching back to main
+
+Can you guess how to switch back to the `main` branch?
+
+What do you think will happen when you do?
+
+What will the history look like?
+
+### Exercise 2 (5 minutes)
+
+Switch back to the `main` branch (hint: The command to run is like the one you used to create a new branch, but you're not creating a new one).
+
+Look at the contents of `mug-brownie.md` - what's changed?
+
+Explore the history with `git log` - make sure you understand the commits in your history and why they're there.
+
+Share with the class what you saw.
+
+### Managing multiple branches
+
+TODO: Talk about how when you create a branch, it will share the history of whatever branch you were on before you created the new one, and that the trainees should generally make sure to branch from `main`.
+
+TODO: Talk about naming branches.
+
+TODO: Explain that we can create multiple branches, and have multiple in-flight pull requests. Benefits: If one idea needs a lot of work it doesn't block the others, reviewer can focus on one thing, keeps changes small.
+
+### Exercise 3 (10 minutes)
+
+Make another branch based on `main` called `extra-topping`.
+
+Add a new topping to the recipe. Make sure to include it both in the ingredients list, and the steps.
+
+Commit your changes.
+
+### Managing branches for pull requests
+
+TODO: Talk through pushing branches to remotes, and making pull requests from them
+
+### Teacher-led example
+
+TODO: Make a pull request from one of your branches. Merge it. Delete the branch. Explain that it's ok to delete the branch because we know the changes are safely on `main`.
```

Comment:
It's not the best formatting of all time but `git log --graph` might be useful here?